calico/felix/plugins/fiptgenerator.py

```diff
@@ -536,15 +546,67 @@ def _build_to_or_from_chain(self, ip_version, endpoint_id, profile_ids,
                 "--match mac ! --mac-source %s" % expected_mac,
                 "Incorrect source MAC"))
 
-        # Jump to each profile in turn.  The profile will do one of the
-        # following:
+        # Tiered profiles come first.
+        deps = set()
+        # Each tier must either accept the packet outright or pass it to the
+        # next tier for further processing.
+        for tier, prof_ids in prof_ids_by_tier.iteritems():
```

Comment:
Hmm, perhaps some general prof/policy renaming in this file?  (I'm guessing this is historical.)